tests/frontend/compilers/test_gaussianmerge.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Unit tests for the GaussianMerge class"""
+
+import pytest
+import numpy as np
+
+import strawberryfields as sf
+import strawberryfields.ops as ops
+
+pytestmark = pytest.mark.frontend
+
+
+def test_complex():
+    modes = 4
+    cutoff_dim = 6
+
+    initial_state = np.zeros([cutoff_dim] * modes, dtype=complex)
+    # The ket below corresponds to a single horizontal photon in each of the modes
+    initial_state[1, 1, 1, 1] = 1
```

Comment:
You could add many other different initial states with low-ish photon number to check things work.

```suggestion

@pytest.mark.parametrize("init", [(1, 1, 1, 1), (0, 2, 1, 0)])
def test_multi_mode_non_gaussian(init):
    modes = 4
    cutoff_dim = 6

    initial_state = np.zeros([cutoff_dim] * modes, dtype=complex)
    # The ket below corresponds to a single horizontal photon in each of the modes
    initial_state[init] = 1
```

Reply:
@federico0112 there's no fixed rule. Obviously more parameterized test cases are always more comprehensive, especially if they are easy to create. Too many cases could be excessive and hinder the readability of the test suite, but 4-6 (like @nquesada suggests above) is a good rule of thumb